### PR TITLE
[SYCL][HIP] Support the SYCL_PI_HIP_MAX_LOCAL_MEM_SIZE environment variable

### PR DESCRIPTION
**Pull request by zjin-lcf**
This PR tries to support the SYCL_PI_HIP_MAX_LOCAL_MEM_SIZE environment variable with the HIP runtime APIs. Thank you for your review.

### CODE REVIEW

**Comment by zjin-lcf** [comment]
Reference
https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#controlling-dpc-cuda-plugin

File: sycl/plugins/hip/pi_hip.cpp [hunk 1]
```diff
@@ -2929,6 +2929,27 @@ pi_result hip_piEnqueueKernelLaunch(
       retImplEv->start();
     }
 
+    // Set local mem max size if env var is present
+    static const char *local_mem_sz_ptr =
+        std::getenv("SYCL_PI_HIP_MAX_LOCAL_MEM_SIZE");
```
Comment (againull) [inline sycl/plugins/hip/pi_hip.cpp:2934]: Could you please add documentation for this environment variable to sycl/doc.

Comment (zjin-lcf) [reply sycl/plugins/hip/pi_hip.cpp:2934]: I added it to sycl/doc. Thanks.

**Review by againull — APPROVED** [review]

File: sycl/plugins/hip/pi_hip.cpp (hunk 1 above)
Comment (againull) [reply sycl/plugins/hip/pi_hip.cpp:2934]: Thanks!

**Review by bader — APPROVED** [review]

**Pull request merged**